packages/open-truss/src/hooks/config-builder.tsx

```diff
@@ -1,37 +1,48 @@
 import get from 'lodash/get'
 import set from 'lodash/set'
-import { createContext, useState } from 'react'
-import { type FrameType, type WorkflowSpec } from '../../configuration'
-import { type YamlType, parseYaml, stringifyYaml } from '../../utils/yaml'
+import { createContext, useCallback, useContext, useState } from 'react'
+import { type FrameType, type WorkflowSpec } from '../configuration'
+import { type YamlType, parseYaml, stringifyYaml } from '../utils/yaml'
 
-export const CONFIG_BASE = `
+const CONFIG_BASE = `
 workflow:
   version: 1
   frameWrapper: ConfigBuilderFrameWrapper
   frames: []
 `.trim()
-export const INITIAL_FRAMES_PATH = 'workflow.frames'
+const INITIAL_FRAMES_PATH = 'workflow.frames'
 
 interface ConfigBuilder {
   config: string
   setConfig: (config: string) => void
   framesPath: string
-  setFramesPath: (path: string) => void
+  setFramesPath: (path: string | null) => void
   addFrame: (frame: FrameType) => void
   deleteFrame: (framePath?: string) => void
 }
-export const ConfigBuilderContext = createContext<ConfigBuilder>({
+
+const ConfigBuilderContext = createContext<ConfigBuilder>({
```

Comment:
FYI I've learned that exporting the result of `createContext` from a package doesn't work well, something to do with types getting weird. Instead we should export a hook (e.g. `useConfigBuilderContext` below) that uses the context.